mesheryctl/internal/cli/root/system/start.go

```diff
@@ -79,174 +85,328 @@ func ValidateComposeFileForRecreation(CurrentServices map[string]utils.Service,
-			//check flag to check successful deployment
-			checkFlag = 0
-			break
-		} else {
-			checkFlag = 1
+	case "kubernetes":
+
+		log.Debug("detecting kubeconfig file...")
+
+		// Detect user's kubeconfig file for the Kubernetes cluster
+		config, err := meshkitkube.DetectKubeConfig()
+
+		if err != nil {
+			return errors.Wrap(err, "failed to detect Kube Config file")
+		}
+
+		log.Debug("creating new Clientset...")
+		// Create a new Clientset for given config
+		clientSet, err := kubernetes.NewForConfig(config)
+
+		if err != nil {
+			return errors.Wrap(err, "error setting clientset")
+		}
+
+		// Create a new client
+		client, err := meshkitkube.New(clientSet, *config)
+
+		if err != nil {
+			return errors.Wrap(err, "failed to create new client")
 		}
-	}
 
-	//if meshery_meshery_1 failed to start showing logs
-	//code for logs
-	if checkFlag == 1 {
-		log.Info("Starting Meshery logging . . .")
-		cmdlog := exec.Command("docker-compose", "-f", utils.DockerComposeFile, "logs", "-f")
-		cmdReader, err := cmdlog.StdoutPipe()
+		log.Debug("fetching required Kubernetes manifest files...")
+		// pick all the manifest files stored in https://github.com/layer5io/meshery/tree/master/install/deployment_yamls/k8s
+		manifests, err := utils.ListManifests(manifestsURL)
+
 		if err != nil {
-			return errors.Wrap(err, utils.SystemError("failed to create stdout pipe"))
+			return errors.Wrap(err, "failed to make GET request")
 		}
-		scanner := bufio.NewScanner(cmdReader)
-		go func() {
-			for scanner.Scan() {
-				log.Println(scanner.Text())
+
+		log.Debug("creating ~/.meshery/manifests folder...")
+		// create a manifests folder under ~/.meshery to store the manifest files
+		if _, err := os.Stat(utils.ManifestsFolder); os.IsNotExist(err) {
+			if err := os.MkdirAll(filepath.Join(utils.MesheryFolder, utils.ManifestsFolder), os.ModePerm); err != nil {
+				return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to make %s directory", utils.ManifestsFolder)))
 			}
-		}()
-		if err := cmdlog.Start(); err != nil {
-			return errors.Wrap(err, utils.SystemError("failed to start logging"))
+			log.Debug("created manifests folder...")
 		}
-		if err := cmdlog.Wait(); err != nil {
-			return errors.Wrap(err, utils.SystemError("failed to wait for command to execute"))
+
+		log.Debugf("downloading manifest files from %s", gitHubFolder)
+
+		// download all the manifest files to the ~/.meshery/manifests folder
+		err = utils.DownloadManifests(manifests, rawManifestsURL)
+
+		if err != nil {
+			return errors.Wrap(err, "failed to download manifests")
+		}
+
+		// downloaded required files successfully now apply the manifest files
+		log.Info("Starting Meshery...")
+
+		log.Debug("applying the manifests to Kubernetes cluster...")
+
+		// apply the adapters mentioned in the config.yaml file to the Kubernetes cluster
+		err = ApplyManifestFiles(RequestedAdapters, client)
+
+		if err != nil {
+			return errors.Wrap(err, "failed to apply manifests")
 		}
 	}
+
```

Comment:
Need default switch case to gracefully exit and inform user that only "docker" and "kubernetes" platforms are supported currently.

Reply:
Yes. I will add that.